modules/heat_conduction/src/actions/ThermalContactAction.C

```diff
@@ -0,0 +1,289 @@
+
+template <>
+InputParameters
+validParams<ThermalContactAction>()
+{
+  InputParameters params = validParams<Action>();
+  params.addClassDescription(
+      "Action that controls the creation of all of the necessary objects for "
+      "calculation of Thermal Contact");
+
+  params.addParam<std::string>(
+      "gap_aux_type",
+      "GapValueAux",
+      "A string representing the Moose object that will be used for computing the gap size");
+  params.addRequiredParam<NonlinearVariableName>("variable", "The variable for thermal contact");
+  params.addRequiredParam<BoundaryName>("master", "The master surface");
+  params.addRequiredParam<BoundaryName>("slave", "The slave surface");
+  params.addParam<Real>("tangential_tolerance",
+                        "Tangential distance to extend edges of contact surfaces");
+  params.addParam<Real>(
+      "normal_smoothing_distance",
+      "Distance from edge in parametric coordinates over which to smooth contact normal");
+  params.addParam<std::string>("normal_smoothing_method",
+                               "Method to use to smooth normals (edge_based|nodal_normal_based)");
+
+  MooseEnum orders(AddVariableAction::getNonlinearVariableOrders());
+  params.addParam<MooseEnum>("order", orders, "The finite element order");
+
+  params.addParam<bool>(
+      "warnings", false, "Whether to output warning messages concerning nodes not being found");
+  params.addParam<bool>(
+      "quadrature", false, "Whether or not to use quadrature point based gap heat transfer");
+
+  params.addParam<std::string>(
+      "appended_property_name", "", "Name appended to material properties to make them unique");
+  params.addRequiredParam<std::string>(
+      "type",
+      "A string representing the Moose object that will be used for heat conduction over the gap");
+  params.addDeprecatedParam<std::vector<VariableName>>(
+      "disp_x", "The x displacement", "Use displacements instead");
+  params.addDeprecatedParam<std::vector<VariableName>>(
+      "disp_y", "The y displacement", "Use displacements instead");
+  params.addDeprecatedParam<std::vector<VariableName>>(
+      "disp_z", "The z displacement", "Use displacements instead");
+  params.addParam<std::vector<VariableName>>(
+      "displacements",
+      "The displacements appropriate for the simulation geometry and coordinate system");
+  params.addParam<std::vector<AuxVariableName>>(
+      "save_in", "The Auxiliary Variable to (optionally) save the boundary flux in");
+
+  params.addParam<Real>("gap_conductivity", 1.0, "The thermal conductivity of the gap material");
+  params.addParam<FunctionName>(
+      "gap_conductivity_function",
+      "Thermal conductivity of the gap material as a function.  Multiplied by gap_conductivity.");
+  params.addParam<std::vector<VariableName>>(
+      "gap_conductivity_function_variable",
+      "Variable to be used in gap_conductivity_function in place of time");
+  params.addParam<std::string>("conductivity_name",
```

Comment:
I realized in my digging around in this code that `conductivity_name` and `conductivity_master_name` don't actually seem to be used by anything. `GapConductance` takes it as an option, but it doesn't use it. You should just get rid of them while you're in here.

Reply:
Great, I'll rip this out. Thanks for the review, and I'll put up a new version asap